inclusive-design/wecount.inclusivedesign.ca
# Fix search functionalities (resolves #562)

## Description

Fixed broken search functionalities provided on the navigation menu (search cross-site) and Views page (search all Views items).

This pull request also takes the chance to fix these issues: 
1. News items are not shown in cross-site search results;
2. The excerpt values are not checked in the search function.

## Steps to test

1. Input a search string in the search box on the navigation bar, then hit Enter;
2. Matched search results should show, including News items;
3. Go to "Views" page, enter a search string in the search box on this page, then hit Enter;
4. Matched search results should show.

**Expected behavior:** 
See above.

## Additional information

The reason of the problem is, when the value of an optional field is not given, this missing value will end up with a line of invalid JSON structure - `"value": ` - in the dumped index.json and viewsWithTags.json. This invalid JSON file is rejected at the search.

The main branch is merged into this PR to include the most recent data that can re-produce this issue.
